### PR TITLE
CDAP-18675 Source parsing in Wrangler

### PR DESCRIPTION
**Pull request by njbriggs**
# CDAP-18675 Source parsing in Wrangler

## Description
Adds source parsing to Wrangler. Source parsing means the source plugin parses the file before it goes to Wrangler, so the user doesn't need to apply `parse-as-csv` or similar directives.
* The dialog will be shown as determined by the connection's response to browse requests. GCS connection currently provides this. Some sources may get this later (e.g. File) and some will never get it (sources that connect to data that is already structured, like BigQuery and databases).
* The selected configuration will be added to the workspace, which will then be passed through to the source plugin in the pipeline.

TODO
- [X] Look at source browse behavior - don't show dialog
- [X] Look at sink behavior - no change needed
- [X] Cypress test updates
- [ ] File encoding not being set correctly - back end fix; shouldn't block this
- [X] Squash commits

## PR Type
- [ ] Bug Fix
- [X] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [CDAP-18675](https://cdap.atlassian.net/browse/CDAP-18675)

## Test Plan
TBD. Will need attention during bug bash

## Screenshots

Config modal:
image

Loading:
image

### CODE REVIEW

File: app/cdap/components/Connections/Browser/GenericBrowser/index.tsx [hunk 1]
```diff
@@ -74,13 +76,16 @@ export function GenericBrowser({ initialConnectionId, onEntityChange, selectedPa
   const [currentConnection, setCurrentConnection] = useState(initialConnectionId);
   const [entities, setEntities] = useState([]);
   const [propertyHeaders, setPropertyHeaders] = useState([]);
+  const [sampleProperties, setSampleProperties] = useState({});
```
Comment (seanfeld) [inline app/cdap/components/Connections/Browser/GenericBrowser/index.tsx:79]: yeesh this file has so many use states - should convert these to a use reducer eventually (not asking for action)

File: app/cdap/components/Connections/Browser/ParsingConfigModal/store.tsx [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    connectionId: connection,
+  };
+
+  const body = {
+    path: entity.path,
+    properties: {},
+  };
+
+  let widgetKey;
+  const widgetObs = ConnectionsApi.getSpecification(params, body).pipe(
```
Comment (seanfeld) [inline app/cdap/components/Connections/Browser/ParsingConfigModal/store.tsx:142]: can we get some context comments

File: app/cdap/components/Connections/Browser/ParsingConfigModal/store.tsx [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+            // set default value
+            if (!selectedProperties[propertyWidget.name] && propertyWidget['widget-attributes']) {
+              selectedProperties[propertyWidget.name] = propertyWidget['widget-attributes'].default;
+            }
+          }
+        });
+      });
+
+      dispatch({
```
Comment (seanfeld) [inline app/cdap/components/Connections/Browser/ParsingConfigModal/store.tsx:192]: can we combine all these into one dispatch call? `SET_INITIAL_PROPERTIES` or something like that and set all the values at once

**Review by seanfeld — APPROVED** [review]
just nits

File: app/cdap/components/Connections/Browser/GenericBrowser/index.tsx (hunk 1 above)
Comment (njbriggs) [reply app/cdap/components/Connections/Browser/GenericBrowser/index.tsx:79]: Acknowledged

**Review by mrahanjam — APPROVED** [review]
LGTM!

**Pull request merged**